### PR TITLE
Model overview: add box plot in the regression case

### PR DESCRIPTION
**Pull request by romanlutz**
## Description

Box plots are so far only there in the binary classification case, but this PR adds a box plot for regression, too! It supports error, predicted y and true y.

image

## Checklist

- [x] I have added screenshots above for all UI changes.
- [x] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: libs/e2e/src/lib/describer/modelAssessment/modelOverview/ensureAllModelOverviewDatasetCohortsViewBasicElementsArePresent.ts [hunk 1]
```diff
@@ -119,9 +119,21 @@ export function ensureAllModelOverviewDatasetCohortsViewBasicElementsArePresent(
         .first()
         .should("have.attr", "aria-label", expectedAriaLabel);
     }
+  } else if (datasetShape.isRegression) {
+    // 2 buttons and 1 hidden overflow button, 3 total
+    cy.get(Locators.ModelOverviewChartPivotItems).should("have.length", 3);
+    cy.get(Locators.ModelOverviewRegressionDistributionChart).should("exist");
+    cy.get(Locators.ModelOverviewProbabilityDistributionChart).should(
+      "not.exist"
```
Comment (imatiach-msft) [inline libs/e2e/src/lib/describer/modelAssessment/modelOverview/ensureAllModelOverviewDatasetCohortsViewBasicElementsArePresent.ts:127]: these kind of constants really should just be in one place... but I guess this is more of a general problem and out of scope for this PR

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+    return (
+      <Stack
+        tokens={{ childrenGap: "10px" }}
+        id="modelOverviewRegressionDistributionChart"
+      >
+        {!noCohortSelected && (
+          <Stack.Item className={classNames.cohortSelectionButton}>
+            <DefaultButton
+              id="modelOverviewRegressionDistributionBoxChartCohortSelectionButton"
```
Comment (imatiach-msft) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx:162]: that's some descriptive id  :laughing:

**Review by imatiach-msft — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx (hunk 2 above)
Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx:162]: You're right, of course, and I dislike it for its length, but I try to err on the side of being descriptive. Coming back to this after a few months I appreciate that it's got all the information on there. 🙂

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+  onBoxPlotStateUpdate: (boxPlotState: IBoxChartState) => void;
+}
+
+interface IRegressionDistributionChartState {
+  targetOption?: IChoiceGroupOption;
+  newlySelectedTargetOption?: IChoiceGroupOption;
+  targetFlyoutIsVisible: boolean;
+}
+
+const targetOptions: IChoiceGroupOption[] = [
```
Comment (vinuthakaranth) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx:43]: Can you please do a fast pass for this newly added plots and buttons?

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+        this.props.cohorts.map((cohort: ErrorCohort, index: number) => {
+          return calculateBoxPlotDataFromErrorCohort(
+            cohort,
+            index,
+            targetOption
+          );
+        })
+      );
+      const outlierData = boxPlotData
+        .map((cohortBoxPlotData) => cohortBoxPlotData?.outliers)
```
Comment (vinuthakaranth) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx:106]: Is the logic to get outlier data is same in all places? can we have one common function to get the data if possible?

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+      if (
+        !_.isEqual(boxPlotData, this.props.boxPlotState.boxPlotData) ||
+        !_.isEqual(this.props.boxPlotState.outlierData, outlierData)
+      ) {
+        this.props.onBoxPlotStateUpdate({ boxPlotData, outlierData });
+      }
+    }
+  }
+
+  public render(): React.ReactNode {
```
Comment (vinuthakaranth) [inline libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx:129]: Please help check reflow scenario with 200% zoom and then 360*256 screen size for newly added plot, thanks~

**Review by vinuthakaranth — APPROVED** [review]

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx (hunk 5 above)
Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx:129]: It scales nicely with smaller size and zoom. 🙂

File: libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx (hunk 4 above)
Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx:106]: I'm unifying it right now. The same can be done for the probability distribution chart for classification. I'm doing that as well.

Comment (romanlutz) [reply libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx]: Done! This turned out to have a ton of potential to unify things. Both chart classes had the box and outlier calculation twice (although not identical, but upon investigating this in more detail it turned out to be easily rewriteable), so I was able to cut 75% of the associated logic 😮

**Pull request merged**